src/helpers/Splitter.ts

```diff
@@ -0,0 +1,82 @@
+import { MetaModel, ReferenceModel, EnumModel, UnionModel, ArrayModel, TupleModel, StringModel, IntegerModel, FloatModel, BooleanModel, ObjectModel, DictionaryModel} from '../models';
+
+export type SplitOptions = {
+  splitEnum?: boolean,
+  splitUnion?: boolean,
+  splitArray?: boolean,
+  splitTuple?: boolean,
+  splitString?: boolean,
+  splitInteger?: boolean,
+  splitFloat?: boolean,
+  splitBoolean?: boolean,
+  splitObject?: boolean,
+  splitDictionary?: boolean
+}
+
+/**
+ * Try split the model
+ * @param model 
+ * @param options 
+ * @param models 
+ * @returns whether the new or old MetaModel to use.
+ */
+// eslint-disable-next-line sonarjs/cognitive-complexity
+const trySplitModel = (model: MetaModel, options: SplitOptions, models: MetaModel[]): MetaModel => {
+  const shouldSplit = options.splitEnum === true && model instanceof EnumModel ||
+    options.splitUnion === true && model instanceof UnionModel ||
+    options.splitArray === true && model instanceof ArrayModel ||
+    options.splitTuple === true && model instanceof TupleModel ||
+    options.splitString === true && model instanceof StringModel ||
+    options.splitInteger === true && model instanceof IntegerModel ||
+    options.splitFloat === true && model instanceof FloatModel ||
+    options.splitBoolean === true && model instanceof BooleanModel ||
+    options.splitObject === true && model instanceof ObjectModel ||
+    options.splitDictionary === true && model instanceof DictionaryModel;
```

Comment:
You can create helper for that:

```ts
function splitGuard(model: MetaModel, condition: boolean, clazz: Object): boolean {
  return condition === true && model instanceof clazz;
}
```

and then in code:

```ts
const shouldSplit = splitGuard(model, options.splitEnum, EnumModel) || ...
```

Reply:
True 😆 

I like the explicit style I think 🤔

Reply:
We can always change it later if we find it weird.